frankenstein/chunk_store.go

```diff
@@ -0,0 +1,379 @@
+package frankenstein
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"net/url"
+	"strings"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/dynamodb"
+	"github.com/aws/aws-sdk-go/service/s3"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/common/log"
+	"github.com/prometheus/common/model"
+
+	"github.com/prometheus/prometheus/storage/metric"
+)
+
+const (
+	hashKey    = "h"
+	rangeKey   = "r"
+	chunkKey   = "c"
+	maxChunkID = "0000000000000000"
+	minChunkID = "FFFFFFFFFFFFFFFF"
+)
+
+var (
+	dynamoRequestDuration = prometheus.NewSummaryVec(prometheus.SummaryOpts{
+		Namespace: "prometheus",
+		Name:      "dynamo_request_duration_seconds",
+		Help:      "Time spent doing DynamoDB requests.",
+	}, []string{"method", "status_code"})
+	dynamoConsumedCapacity = prometheus.NewCounterVec(prometheus.CounterOpts{
+		Namespace: "prometheus",
+		Name:      "dynamo_consumed_capacity",
```

Comment:
Counters should always end in `_total` (a convention that helps distinguish them from gauges in situations where it could be unclear).